contrib/golang/filters/http/source/golang_filter.h

```diff
@@ -193,6 +192,21 @@ class Filter : public Http::StreamFilter,
                               std::function<void(Http::ResponseHeaderMap& headers)> modify_headers,
                               Grpc::Status::GrpcStatus grpc_status, absl::string_view details);
 
+  absl::optional<absl::string_view> getHeaderInternal(absl::string_view key)
+      ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_);
```

Comment:
Again, it's most important to have the annotations on the data members themselves. Please do that.

/wait

Reply:
> Again, it's most important to have the annotations on the data members themselves. Please do that.
> 
> /wait
@mattklein123 Because the member does not need to be locked in some scenarios,  the `ABSL_GUARDED_BY` annotation cannot be added to the member. So use `ABSL_EXCLUSIVE_LOCKS_REQUIRED` to annotate in concurrent methods. Such as below

https://github.com/envoyproxy/envoy/blob/1ca4641ccbb7edf0dec3da7ecf22d8400e52f5e5/contrib/golang/filters/http/source/golang_filter.cc#L586-L590

Reply:
For right now instead of trying to over-optimize I would just lock in all cases with annotations. It's safer and after things are proven to be stable we can relax later if needed.

/wait